XMonad/Layout/IndependentScreens.hs

```diff
@@ -101,13 +110,30 @@ unmarshallW = snd . unmarshall
 workspaces' :: XConfig l -> [VirtualWorkspace]
 workspaces' = nub . map unmarshallW . workspaces
 
+-- | Specify workspace names for each screen
+withScreen :: ScreenId            -- ^ The screen to make workspaces for
+           -> [VirtualWorkspace]  -- ^ The desired virtual workspace names
+           -> [PhysicalWorkspace] -- ^ A list of all internal physical workspace names
+withScreen n vws = [marshall n pws | pws <- vws]
+
+-- | Make all workspaces across the monitors bear the same names
 withScreens :: ScreenId            -- ^ The number of screens to make workspaces for
             -> [VirtualWorkspace]  -- ^ The desired virtual workspace names
             -> [PhysicalWorkspace] -- ^ A list of all internal physical workspace names
-withScreens n vws = [marshall sc pws | pws <- vws, sc <- [0..n-1]]
+withScreens n vws = concatMap (`withScreen` vws) [0..n-1]
 
 onCurrentScreen :: (VirtualWorkspace -> WindowSet -> a) -> (PhysicalWorkspace -> WindowSet -> a)
-onCurrentScreen f vws = screen . current >>= f . flip marshall vws
+onCurrentScreen f vws = W.screen . W.current >>= f . flip marshall vws
+
+-- | 'currentWorkspaces' takes a 'PhysicalWindowSpace' sorting function, the workspaces on 
+-- current screen and returns the sorted results. Can be used by other integration functions later.
+currentWorkspaces :: ([PhysicalWindowSpace] -> [PhysicalWindowSpace]) -> X [PhysicalWorkspace]
```

Comment:
What's the intended use of this function? Is it still useful or was that part of some integration stuff that was since removed from this PR?

Reply:
Yeah, I intend to use it for later integration stuff. Right now the integration stuffs are removed from this PR.

Reply:
In that case it would be best to drop this from the PR as well, it seems out of place in a "fix #420" PR.